pkg/storage/fingerprint_writer.go

```diff
@@ -49,8 +51,13 @@ func makeFingerprintWriter(
 	hasher hash.Hash64,
 	cs *cluster.Settings,
 	f io.Writer,
+	startKey roachpb.Key,
 	opts MVCCExportFingerprintOptions,
-) fingerprintWriter {
+) (fingerprintWriter, error) {
+	_, tenantPrefix, err := keys.DecodeTenantPrefix(startKey)
+	if err != nil {
+		return fingerprintWriter{}, err
+	}
```

Comment:
I believe we are decoding this so that we can use StripTenantPrefix below which asserts that the prefix is the same.  But what if I want to use this from the system tenant on a span that crosses tenant boundaries. Are we saying that isn't a valid use case. If so, we should include some reasoning for why not.

Reply:
yeah, this is a good point. Though I wonder if using a codec's `StipTenantPrefix` is more performant than `DecodeTenantPrefix`. I can run some benchmarks.

Reply:
there's no noticeable perf difference between `StripTenantPrefix` vs `DecodeTenantPrefix`, according to a macro benchmark (fingerpinting a 15 gb table). So I suppose I shouldn't add this restriction to limit the cmd to a single tenant key span -- i'll need to create a slightly more complicated helper to strip the index prefix then.